telemetry-core/src/main/java/com/newrelic/telemetry/metrics/MetricBuffer.java

```diff
@@ -39,12 +40,36 @@ public MetricBuffer(Attributes commonAttributes) {
   }
 
   /**
-   * Append a {@link Metric} to this buffer, to be sent in the next {@link MetricBatch}.
+   * Append a {@link Count} to this buffer, to be sent in the next {@link MetricBatch}.
+   *
+   * @param countMetric The new {@link Count} instance to be sent.
+   */
+  public void addMetric(Count countMetric) {
```

Comment:
Could you help me understand why the three addMetrics with specific parameter types instead of the original one?

Reply:
This was done in anticipation of the batching change (since we need to know the exact event type to generate the JSON).

Reply:
Right, because each Metric type has a different format